src/app/acorns/acorns.controller.js

```diff
@@ -5,56 +5,53 @@
  * Created by vytautassugintas on 06/04/16.
  */
 (function () {
-    'use strict';
-    var AcornController = function ($http, $scope, $window, $cookies, $timeout, $httpParamSerializer, $filter, ProfileService, Challenges, Resources) {
-
-        $scope.greeting = "hello";
-
-        $scope.showLoader = true;
-
-        $scope.userAvailableKudos = 0;
-        $scope.buttonDisabled = true;
-
-        $scope.receivedAcorns = false;
-        $scope.sentAcorns = true;
-
-        $scope.isValid = isValid;
-        $scope.checkUser = checkUser;
-
-        checkUser();
-
-        ProfileService.remainingKudos().then(function (val) {
-            Resources.setUserAvailableKudos(val);
-        });
-
-        ProfileService.receivedKudos().then(function (val) {
-            $scope.userReceivedKudos = val;
-            $scope.showLoader = false;
-        });
-
-        $scope.$watch(function () {
-            return Resources.getUserAvailableKudos()
-        }, function (newVal) {
-            if (!isValid(newVal)) $scope.userAvailableKudos = Resources.getUserAvailableKudos();
-        });
-
-        function isValid(value) {
-            return typeof value === "undefined";
-        }
-
-        function checkUser() {
-            ProfileService.checkUser().then(function (val) {
-                val.logged ? $window.location.href = "#/acorns" : $window.location.href = "#/login";
+        'use strict';
+        var AcornController = function ($http, $scope, $window, $cookies, $timeout, $httpParamSerializer, $filter, ProfileService, Challenges, Resources) {
+            $scope.showLoader = true;
+
+            $scope.isValid = isValid;
+            $scope.checkUser = checkUser;
+
+            checkUser();
```

Comment:
It's common to have one `activate()` function in a controller/directive where all the initialization logic is contained, so other developers know where everything is initialized. `angular.component` even has an `$onInit` lifecycle hook exactly for that reason. It's probably a good idea to move this into the `activate()` function so all the init logic is located in one place.